packages/project/package.json

```diff
@@ -18,12 +18,12 @@
     "@webstudio-is/react-sdk": "*",
     "bson-objectid": "^2.0.2",
     "immer": "^9.0.12",
-    "nanoid": "^3.2.0",
     "slugify": "^1.6.5",
     "uuid": "^9.0.0",
     "zod": "^3.19.1"
   },
   "devDependencies": {
+    "nanoid": "^3.2.0",
```

Comment:
This is weird-ish and shouldn't be required. However, `nanoid` is an ESM-only package and as such, it can't be required. So we need to bundle it into `@webstudio-is/project` with `tsup` (and that's why it has been moved from deps to devDeps). This way Remix won't try to `require` it.

Reply:
omg .... there must be some generic solution to that cc @TrySound

Reply:
problem is its going to confuse people for being in devDependencies and they will potentially move it back to dependencies

Reply:
ESM-only packages tend to be problematic in the ecosystem. This is even called out by the Remix docs here:
https://remix.run/docs/en/v1/pages/gotchas#importing-esm-packages

This thing here was somewhat required by the other change that I've made to `serverDependenciesToBundle`:
https://github.com/webstudio-is/webstudio-designer/pull/383#discussion_r1014602193

Because now I don't bundle the `@webstudio/*` packages I can't just bundle something that they require - it's impossible. 

Perhaps I should try to figure out why `sharp` has started to be bundled now instead but I think it's better to land this to fix your immediate issues and try to investigate this further separately. Once I would get to the bottom of the sharp thing we could revert this change.

Note though that I stand by the comment made about the `serverDependenciesToBundle`. Usually, there shouldn't be any reason to bundle your monorepo packages like this. I suspect this was already a bandaid for the problems like this one - consuming ESM-only packages deeper in the dep graph etc. So while conceptually it shouldn't have to be done, there might be some technical limitation like this that might make it a lesser evil.

It's also worth noting that by relying on ESM-only packages your packages are currently a little bit broken anyway (if they would be consumed from npm). Your packages are at the moment CJS-only and that is incompatible with ESM-only packages.

Reply:
> problem is its going to confuse people for being in devDependencies and they will potentially move it back to dependencies

Sure, I agree with that. That's why I'm proposing to land this as a stopgap solution - if it doesn't break anything for your right now - and improve that soon.

Reply:
> @webstudio/*

There was no way using a different matcher than this? one that understands .server?

Reply:
Hmm, those are somewhat unrelated. Your server can still import regular non-.server files and you could end up with the same problem.

To work around the esm-only issue and to continue bundling your monorepo packages for your server bundles we'd like to:
- use `serverDependenciesToBundle: ['@webstudio/*']`
- control external and put `sharp` there. However, Remix doesn't allow you to do that. **Maybe**, you can declare `sharp` as a dependency of your designer app and that would force Remix to infer it as being external. This could be confusing as well though because your Remix app doesn't actually directly depend on `sharp` so it would look redundant. 

In Remix it seems that the bundle/not-bundle decision for the server entry is taken [here](https://github.com/remix-run/remix/blob/3832d457b34d7a62633bf86a2606971263e710cb/packages/remix-dev/compiler/plugins/serverBareModulesPlugin.ts#L15-L19). So I could investigate this. On the other hand... once you start bundling some stuff I really think that you should just go all in. I'd have to think about it more to assess the exact risks but my hot take is that "partial bundling" can almost always manifest in some obscure bugs. They might be edge cases and you might never encounter them but that doesn't mean that they are impossible.

Reply:
Oh I don't care if we bundle something or not, esbuild is fast enough. I only care that we 
1. don't add serverside stuff to the client
2. have normal way of defining dependencies

Reply:
Do you want any of those fixed as part of this PR? Or can I investigate those in followups?

Reply:
up to you, I would try and bundle everything with esbuild so we never ever have to worry about esm vs cjs

Reply:
Lets create a follow-up issue if that's better